bundle/jsky.app.ot/src/main/java/jsky/app/ot/editor/seq/DynamicSequenceTableModel.java

```diff
@@ -40,22 +36,22 @@
 
     private SPNodeKey _nodeKey;
 
-    public void setSequence(ConfigSequence sequence, SPNodeKey key, List<ItemKey> alwaysShow) {
+    public void setSequence(final ConfigSequence sequence, final SPNodeKey key, final List<ItemKey> alwaysShow) {
         _sequence     = sequence;
         _nodeKey      = key;
 
         final ItemKey[] newIteratedKeys = sort(getIteratedKeys(sequence, alwaysShow));
-        if (newIteratedKeys.equals(_iteratedKeys)) {
+        if (Arrays.equals(newIteratedKeys, _iteratedKeys)) {
             fireTableDataChanged();
         } else {
-            _iteratedKeys = sort(getIteratedKeys(sequence, alwaysShow));
+            _iteratedKeys = newIteratedKeys;
             fireTableStructureChanged();
         }
     }
 
-    private Set<ItemKey> getIteratedKeys(ConfigSequence seq, List<ItemKey> alwaysShow) {
-        Set<ItemKey> res = new HashSet<ItemKey>();
-        ItemKey[]   keys = seq.getIteratedKeys();
+    private Set<ItemKey> getIteratedKeys(final ConfigSequence seq, final List<ItemKey> alwaysShow) {
+        final Set<ItemKey> res = new HashSet<>();
+        final ItemKey[]   keys = seq.getIteratedKeys();
         for (ItemKey key : keys) {
             // Strip out all the smart gcal metadata.
             if (MetaDataConfig.MATCHER.matches(key) || SmartgcalSysConfig.MATCHER.matches(key)) continue;
```

Comment:
Seems like this could be done with a `filter` and `foreach`. Not sure if that is more readable, though.
`Arrays.stream(seq.getIteratedKeys())
.filter(key -> !(MetaDataConfig.MATCHER.matches(key) && SmartgcalSysConfig.MATCHER.matches(key)) && templ.matches(key))
.foreach(res::add);`